File: cloud/oauth-server/src/main/java/pers/acp/admin/oauth/repo/UserRepository.java
package pers.acp.admin.oauth.repo;

import pers.acp.admin.oauth.base.OauthBaseRepository;
import pers.acp.admin.oauth.entity.User;

import java.util.List;
import java.util.Optional;

/**
 * @author zhangbin by 2018-1-17 17:48
 * @since JDK 11
 */
public interface UserRepository extends OauthBaseRepository<User, String> {

    Optional<User> findByLoginno(String loginno);

    Optional<User> findByMobile(String mobile);

    Optional<User> findByLoginnoAndIdNot(String loginno, String userId);

    Optional<User> findByMobileAndIdNot(String mobile, String userId);

    List<User> findByLevelsGreaterThan(int currLevels);

    void deleteByIdIn(List<String> idList);

}

File: cloud/oauth-server/src/main/kotlin/pers/acp/admin/oauth/repo/UserRepository.kt
package pers.acp.admin.oauth.repo

import pers.acp.admin.common.base.BaseRepository
import pers.acp.admin.oauth.entity.User
import java.util.*

/**
 * @author zhangbin by 2018-1-17 17:48
 * @since JDK 11
 */
interface UserRepository : BaseRepository<User, String> {

    fun findByLoginNo(loginNo: String): Optional<User>

    fun findByMobile(mobile: String): Optional<User>

    fun findByLoginNoAndIdNot(loginNo: String, userId: String): Optional<User>

    fun findByMobileAndIdNot(mobile: String, userId: String): Optional<User>

    fun findByLevelsGreaterThan(currLevels: Int): MutableList<User>

    fun findByLoginNoLikeOrNameLikeOrderByLoginNoAsc(loginNo: String, name: String): MutableList<User>

    fun deleteByIdIn(idList: MutableList<String>)

}
